core/lib/dal/src/sync_dal.rs

```diff
@@ -35,6 +35,8 @@ impl SyncDal<'_, '_> {
                             (MAX(number) + 1)
                         FROM
                             l1_batches
+                        WHERE
+                            is_sealed
```

Comment:
[Out of this PR scope]
As you can see the pattern `COALESCE(miniblocks.l1_batch_number, SELECT (MAX(number) + 1) FROM l1_batches WHERE is_sealed)` is quite common. We can consider saving l1 batch number when inserting miniblock. This should simplify such places. Though we should be careful, AFAIR we use `miniblocks.l1_batch_number IS NULL` to get miniblocks in pending batch (e.g. to re-execute them in state-keeper). We can replace it with `SELECT ... FROM miniblocks JOIN l1_batches ON miniblocks.l1_batch_number = l1_batch.number WHERE l1_batch.is_sealed = FALSE` but I don't know how indexing will work for such query.
Please investigate indexing and consider this option

Reply:
That is indeed be the next planned step after PLA-1011. It will simplify some code, also this is somewhat of a weird pattern as some node operators are confused when `miniblocks.l1_batch` is set to null when we actually know which batch it is supposed to belong to.